docs/developers/talawa/flutter_testing_guide.md

```diff
@@ -0,0 +1,274 @@
+  ```
+  
+  Now, you can use the `calculator` mock object to simulate the behavior of the `Calculator` in your tests. For example, here's a test that verifies that the
+  `CalculatorController` correctly adds two numbers:
+  
+  ```dart
+  test('CalculatorController adds two numbers', () {
+  // Create a new CalculatorController, passing in the mock Calculator
+  final controller = CalculatorController(calculator);
+
+  // Stub the add method on the mock calculator
+  when(calculator.add(2, 3)).thenReturn(5);
+
+  // Call the addNumbers method on the controller
+  final result = controller.addNumbers(2, 3);
+
+  // Verify that the add method was called on the calculator
+  verify(calculator.add(2, 3)).called(1);
+
+  // Verify that the result returned by the controller is correct
+  expect(result, equals(5));
+});
+  ```
+  
+  In this test, the `MockCalculator` is created and passed to the `CalculatorController`. The `when` method is used to stub the `add` method on the mock calculator, so
+  that when the `add` method is called with arguments 2 and 3, it returns the value 5. Then, the `addNumbers` method is called on the `CalculatorController`, and the
+  result is verified using the `expect` method. Finally, the `verify` method is used to ensure that the `add` method was called on the mock calculator with the correct
+  arguments.
+  
+    
+### Stubs
+
+Stubbing is a technique used in testing to replace a real object with a simplified version that provides predictable behavior.
+
+In Flutter, you can use stubs to replace real objects with mock objects or other simplified versions.
+
+To stub a method or class in Flutter, you can use the when() function provided by the `mockito` package. For example, if you have a method called myMethod() that you want to stub, you can do the following:
+
+```dart
+var myMock = MockMyObject();
+when(myMock.myMethod()).thenReturn('my result');
+```
+This will replace the `myMethod()` method on the `myMock` object with a stub that always returns the string `my result`.
+
+`when()` method is used in the previous Calculator example as well where it is used to stub the add method to mock the Calculator.
+
+You can also use the any matcher to match any input value. For example:
+
+```dart
+when(myMock.myMethod(any)).thenReturn('my result');
+```
+This will stub the `myMethod()` method to always return `my result`, regardless of the input value.
+
+# Example Test of Mocks and Stubs
+
+There is a function in talawa which provides chat service in the app. The file is located in `talawa/lib/services/chat_service.dart`. Unit Tests for this file have been written to test all the methods written in the file for instance `sendMessageToDirectChat` and `getDirectChatsByUserId` methods. These tests are written in the file `talawa\test\service_tests\chat_service_test.dart`. These tests have all the necessary mocks and stubs to understand the concept of testing in talawa.
+
+# sendMessageToDirectChat Method
+
```

Comment:
Add introductory text that states that in this example we'll be testing the `sendMessageToDirectChat` method. It makes the flow better.